src/layout/workspace.rs

```diff
@@ -261,9 +264,23 @@ impl From<PresetWidth> for ColumnWidth {
 
 impl<W: LayoutElement> Workspace<W> {
     pub fn new(output: Output, options: Rc<Options>) -> Self {
+        Self::new_with_config(output, None, options)
+    }
+
+    pub fn new_with_config(
+        output: Output,
+        config: Option<&WorkspaceConfig>,
```

Comment:
For consistency with the method below, I'd make this owned. If we add more config stuff in the future I can see just storing this struct in the `Workspace`, which will make this need an owned value even more.
```suggestion
        config: Option<WorkspaceConfig>,
```